### PR TITLE
Issue #3136 ATC Check - GEN_INCLUDE_NOT_EXEC dump

### PR DESCRIPTION
**Pull request by jrodriguez-rc**
Fix for issue #3136 ATC Check - GEN_INCLUDE_NOT_EXEC dump

### CODE REVIEW

File: src/zcl_abapgit_code_inspector.clas.abap [hunk 1]
```diff
@@ -32,6 +32,11 @@ CLASS zcl_abapgit_code_inspector DEFINITION
         !io_set TYPE REF TO cl_ci_objectset
       RAISING
         zcx_abapgit_exception .
+    METHODS check_obj_before_add
```
Comment (larshp) [inline src/zcl_abapgit_code_inspector.clas.abap]: suggest renaming to "skip_object" takes object as input, returns true if it is an object to be skipped

Comment (jrodriguez-rc) [reply src/zcl_abapgit_code_inspector.clas.abap]: I changed it.
In the new method, I create a CASE for possible future cases

File: src/zcl_abapgit_code_inspector.clas.abap [hunk 2]
```diff
@@ -217,14 +217,14 @@ CLASS zcl_abapgit_code_inspector IMPLEMENTATION.
           INTO ls_trdir
           FROM trdir
           WHERE name = is_obj-objname.
-        IF ls_trdir-subc = 'I'. " Include program
+        IF ls_trdir-subc <> 'I'. " Include program
           RETURN.
```
Comment (larshp) [inline src/zcl_abapgit_code_inspector.clas.abap]: ideally `rv_skip` should be set in all program paths, not relying on initial field values

Comment (jrodriguez-rc) [reply src/zcl_abapgit_code_inspector.clas.abap]: Sorry but I don't understand what you mean about program paths and initial field values

Comment (larshp) [reply src/zcl_abapgit_code_inspector.clas.abap]: method can `RETURN` without setting a value for `rv_skip`, plus with RETURN the control flow does not end at ENDMETHOD

if
```abap
        IF ls_trdir-subc <> 'I'. " Include program
          RETURN.
        ENDIF.

        rv_skip = abap_true.
```
is replaced with
```abap
        rv_skip = boolc( ls_trdir-subc = 'I' ). " Include program
```

then all control flow will end at ENDMETHOD, and rv_skip is always set, no implicit return of initial values

Comment (jrodriguez-rc) [reply src/zcl_abapgit_code_inspector.clas.abap]: I changed the logic and I've had to replace CASE for an IF to leave method most clear possible, so checks won't allow a case with one only WHEN

File: src/zcl_abapgit_code_inspector.clas.abap [hunk 3]
```diff
@@ -183,9 +190,37 @@ CLASS ZCL_ABAPGIT_CODE_INSPECTOR IMPLEMENTATION.
...
+      p_objects = lt_objs_check ).
+
+  ENDMETHOD.
+
+
+  METHOD skip_object.
+
+    DATA: ls_trdir TYPE trdir.
+
+    IF is_obj-objtype =  'PROG'.
```
Comment (larshp) [inline src/zcl_abapgit_code_inspector.clas.abap]: IF is_obj-objtype <> 'PROG', then variable `rv_skip` is not set, relies on implicit initial value according to the type

Comment (jrodriguez-rc) [reply src/zcl_abapgit_code_inspector.clas.abap]: I set again CASE to set abap_false in WHEN OTHERS.
I thought don't care if `rv_skip` was abap_false or an initial value

**Pull request merged**